test/StringToolsTests.h

```diff
@@ -0,0 +1,333 @@
+                   false,StringTools::PatternMatch(SecondSource,FourthPattern,true));
+        TEST_EQUAL("PatternMatch(const_StringView,const_StringView,const_Boole)-Fail3",
+                   false,StringTools::PatternMatch(FirstSource,ThirdPattern,false));
+        TEST_EQUAL("PatternMatch(const_StringView,const_StringView,const_Boole)-Fail4",
+                   false,StringTools::PatternMatch(ThirdSource,SixthPattern,true));
+    }//Pattern Matching
+
+    {//Convert-From-String Functions
+        TEST_EQUAL_EPSILON("ConvertFromString(const_StringView)-Real",
+                   Real(3.1415926),StringTools::ConvertFromString<Real>("3.1415926"));
+        TEST_EQUAL("ConvertFromString(const_StringView)-Integer",
+                   Integer(-7272),StringTools::ConvertFromString<Integer>("-7272"));
+        TEST_EQUAL("ConvertFromString(const_StringView)-Whole",
+                   Whole(8383),StringTools::ConvertFromString<Whole>("8383"));
+        TEST_EQUAL("ConvertFromString(const_StringView)-Int8",
+                   static_cast<Int8>(-45),StringTools::ConvertFromString<Int8>("-45"));
+        TEST_EQUAL("ConvertFromString(const_StringView)-UInt8",
+                   static_cast<UInt8>(66),StringTools::ConvertFromString<UInt8>("66"));
+        TEST_EQUAL("ConvertFromString(const_StringView)-Int16",
+                   Int16(-30100),StringTools::ConvertFromString<Int16>("-30100"));
+        TEST_EQUAL("ConvertFromString(const_StringView)-UInt16",
+                   UInt16(27015),StringTools::ConvertFromString<UInt16>("27015"));
+        TEST_EQUAL("ConvertFromString(const_StringView)-Int32",
+                   Int32(-1256844),StringTools::ConvertFromString<Int32>("-1256844"));
+        TEST_EQUAL("ConvertFromString(const_StringView)-UInt32",
+                   UInt32(3872423),StringTools::ConvertFromString<UInt32>("3872423"));
+
+        TEST_EQUAL("ConvertToBool(const_StringView,const_Boole)-Yes",
+                   true,StringTools::ConvertToBool("Yes",false));
+        TEST_EQUAL("ConvertToBool(const_StringView,const_Boole)-No",
+                   false,StringTools::ConvertToBool("No",true));
+        TEST_EQUAL("ConvertToBool(const_StringView,const_Boole)-1",
+                   true,StringTools::ConvertToBool("1",false));
+        TEST_EQUAL("ConvertToBool(const_StringView,const_Boole)-0",
+                   false,StringTools::ConvertToBool("0",true));
+        TEST_EQUAL("ConvertToBool(const_StringView,const_Boole)-True",
+                   true,StringTools::ConvertToBool("True",false));
+        TEST_EQUAL("ConvertToBool(const_StringView,const_Boole)-False",
+                   false,StringTools::ConvertToBool("False",true));
+        TEST_EQUAL("ConvertToBool(const_StringView,const_Boole)-Default",
+                   true,StringTools::ConvertToBool("Yes"));
+        TEST_EQUAL("ConvertToBool(const_StringView,const_Boole)-Gibberish",
+                   false,StringTools::ConvertToBool("Gibberish"));
```

Comment:
I think a ("gibberish", true) test should be added to specifically test the defaulting behavior.

Reply:
Added.